### ISSUE TITLE
May I put my work of CRDT-Redis here? It is a prototype for my research.

### ISSUE THREAD

**Issue by elem-azar-unis** [redis/redis-doc#1679]
Hi, I implemented my version of CRDT into Redis. It is called [CRDT-Redis](https://github.com/elem-azar-unis/CRDT-Redis).

There are some CRDTs currently implemented:

* Replicated Priority Queue (RPQ)
  * [Add-Win RPQ](document/add-win-crpq.pdf)
  * [Remove-Win RPQ](document/rwf-tr.pdf)
  * [RWF-RPQ](document/rwf-tr.pdf)
* List
  * [Remove-Win List](document/rwf-tr.pdf)
  * [RWF-List](document/rwf-tr.pdf)

It is a prototype system that is mainly for my research of designing, implementing, and verifying CRDTs. Some parts of the system are implemented quite simply, such as the P2P replication mode, and I didn't consider the persistence of the CRDTs I added.

Is there a suitable place to put my work here? Any chance it could be shown on the redis.io website?

**Comment by itamarhaber** [comment]
Hi @elem-azar-unis 

This sounds like a nice research project, well done!

Firstly, there's no license in the project, so it is totally unclear what one can do with it. That said, it would conceptually belong under https://redis.io/documentation#other-distributed-systems-based-on-redis.

However, I'm unclear whether the project provides real value to users (on top of its research and inspirational values). I feel uncomfortable putting it on display without some assurance (to the users) that it is usable and maintained.

**Comment by elem-azar-unis** [comment]
Hi @itamarhaber 

I‘ve added the license. My mistake.

I must admit that it is mainly a research prototype. I focused on the CRDT part and simplified many things. The main shortages I suppose are the simplified P2P replication and the lack of persistence of the CRDTs. However, it is useful for my research. I believe it could be useful for anyone who wants to try out the Redis with CRDT.

The CRDT-Redis ran for days when I gathered experimental data for my research. The performance was fair. I tested some of the CRDTs with test cases generated by TLA+ models. So I'm quite confident with that part of the system.

Still, there is some work needed for the system to be more useful. Industrially useful as I think.  P2P replication, the persistence of the CRDTs, and more  CRDT operations (mainly read operations). But to be honest, this is a bit off my goal. Perhaps those who pursue more can continue to develop it from here 🤔 

The place https://redis.io/documentation#other-distributed-systems-based-on-redis? Wow, Redis Enterprise and Roshi are here? My project ... Could it really be put beside them 🤣
